xapian-applications/omega/templates/query

```diff
@@ -106,7 +106,7 @@ $if{$field{size},<span title="$html{$field{size}} bytes">Size: <b>$html{$filesiz
 </div>
 </div>
 </td>
-<td><b><a href="$html{$field{url}}">$html{$or{$field{caption},$field{title},$prettyurl{$field{url}},Untitled}}</a></b><br>
+<td><b><a href="$html{omega.cgi?FMT=$url{clicklog}&QUERYID=$url{$hash{md5,$transform{s/^\s+|\s+$$//g,\1,$query}}}&DOCID=$url{$id}&URL=$url{$field{url}}}">$html{$or{$field{caption},$field{title},$prettyurl{$field{url}},Untitled}}</a></b><br>
```

Comment:
You don't have to use an IP address, because you don't need to identify users: you can identify the _session_. In this case it's a query session, which starts when you search, and continues while you're conducting the same search. So you want to create a new session id, which continues as you page through search results. The fact that in theory someone could copy that URL and pass it to someone else is probably worth documenting and ignoring (if someone wanted to do anything more "serious", using ip addresses, they can but for privacy reasons they should set that up themselves because only they can understand the consequences under their local legal system).

If you look at CGI parameters such as `xP`, you'll see that there's logic for figuring out if the search has changed. I believe (although I haven't checked quite how it does this) that internally to omega this gets assessed and used to reset things (relevance judgements and the page number, for instance), and would provide a hook for changing the session id.

It's fine to leave this for now if you don't need it for your click modelling, but it's worth making a note of this as a possible future direction.